tests/custom_resp_header_spec.rs

```diff
@@ -37,4 +32,17 @@ mod integration_tests {
     assert_eq!(response_headers.get("x-id").unwrap(), "1");
     assert_eq!(response_headers.get("x-value").unwrap(), "value");
   }
+
+  #[tokio::test]
+  async fn test_verify_response_headers() {
+    let schema_path = "tests/graphql_mock/test-custom-headers.graphql";
+    verify_response_headers(schema_path).await;
+  }
+
+  #[tokio::test]
+  async fn test_verify_response_headers_over_http() {
+    let schema_path =
+      "https://raw.githubusercontent.com/tailcallhq/tailcall/main/tests/graphql_mock/test-custom-headers.graphql";
+    verify_response_headers(schema_path).await; //might fail while testing without internet
+  }
```

Comment:
I think we should add tests for the reader insider of `reader.rs`.

Reply:
@tusharmath it should be placed in tests/reader.rs as we'll need files to test IO and hardcoding example/* will be a problem.

Reply:
@ssddOnTop Why is this spec modified? Its only testing that custom headers set in the config file are set in the response. Its not related to how the config file is loaded.

Reply:
@sujeetsr I don't think it was necessary as it was already being tested by local file path. Running the tests with github url just rate limits the developer